OpenRA.Mods.Common/UpdateRules/Rules/20221203/AddColorPickerValueRange.cs

```diff
@@ -0,0 +1,43 @@
+﻿#region Copyright & License Information
+/*
+ * Copyright (c) The OpenRA Developers and Contributors
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System.Collections.Generic;
+
+namespace OpenRA.Mods.Common.UpdateRules.Rules
+{
+	public class AddColorPickerValueRange : UpdateRule
+	{
+		public override string Name => "V was removed and PresetValues were added to ColorPickerManager.";
+
+		public override string Description =>
+			"Each preset color now needs to have brightness specified";
+
+		public override IEnumerable<string> UpdateActorNode(ModData modData, MiniYamlNode actorNode)
+		{
+			var manager = actorNode.LastChildMatching("ColorPickerManager");
+			if (manager == null)
+				yield break;
+
+			var v = manager.LastChildMatching("V")?.NodeValue<float>() ?? 0.95f;
+			manager.RemoveNodes("V");
+
+			var length = manager.LastChildMatching("PresetHues")?.NodeValue<float[]>().Length;
+			if (length == null)
+				yield break;
+
+			var vrray = new float[length.Value];
```

Comment:
```suggestion
			var vArray = new float[length.Value];
```
(But this is in the update rule, so not important.)